### PR TITLE
Gco rans3p fix

### PR DESCRIPTION
**Pull request by cekees**
@manuel-quezada @Giovanni-Cozzuto-1989 @adimako there were quite a few issues with the three-phase model. This version is working with all the optimized modules with both fully closed and open boundaries on the standard dambreak problem. @Giovanni-Cozzuto-1989's sediment test problem sort of does the right thing, but my understanding is that @Giovanni-Cozzuto-1989 had commented out some of the closure relations that prevent the model from being physically realistic (e.g. the interphase momentum transfer/drag terms are commented out and the sediment stress tensor is not enabled).  For this reason the sediment accelerates rapidly in the direction of gravity and there is no physical mechanism to prevent the sediment volume fraction going above the critical dense packing and in fact to 1, in which case many of the equations become singular because all their terms are multiplied by porsoity=1-vos. I think the next step is for @Giovanni-Cozzuto-1989 to try this branch, then possibly merge with his (as this PR is setup to do), then re-enable the drag terms, the enable the sediment stress tensor. We're going to need to add the granular temperature balance equation as a new model.

### CODE REVIEW

**Comment by Giovanni-Cozzuto-1989** [comment]
Yes, that is exactly what I did.
@cekees do you agree with this version of the penalty term then?
I mean with the vos scaling the density at:
- https://github.com/erdc/proteus/blob/5d031536e72a37696151cff7249a3dad71374c66/proteus/mprans/PresInc.py#L190 
- https://github.com/erdc/proteus/blob/5d031536e72a37696151cff7249a3dad71374c66/proteus/mprans/PresInc.h#L147

**Comment by Giovanni-Cozzuto-1989** [comment]
@cekees @adimako @tridelat 
I keep on having this error message while this branch is compiling on my computer:

... : warning: unused variable ...

This is happening mainly in RANS3PF.h (probably will be the same for RANS3PF2D.h) and at the moment I'm switching OFF those variables. This fix the problem but then other ones break my installation again.
Do you know what I should do?

**Comment by adimako** [comment]
Resolved some conflicts as gco's version has the Mint terms switched on

**Comment by adimako** [comment]
@Giovanni-Cozzuto-1989 technically it is a warning and it allows the code to be compiled. I think @cekees had inserted some test variables that do not affect the code. I will try to clean these up

**Comment by tridelat** [comment]
@Giovanni-Cozzuto-1989 These warning are just for variables that were declared but are not actually used in the code, so it should not be a problem to leave them, even though it is cleaner to remove them if they are not necessary

**Comment by adimako** [comment]
@cekees @Giovanni-Cozzuto-1989 this one failed because there was some code left hanging after the conflicts were resolved. I will fix this later and ask for a review of the particular block

**Comment by adimako** [comment]
@Giovanni-Cozzuto-1989 this should be alright now, can you review?

File: proteus/mprans/RANS3PSed.py [hunk 1]
```diff
@@ -1676,7 +1677,7 @@ def __init__(self,
                 ('u', 2)] = self.numericalFlux.ebqe[
                 ('u', 1)].copy()
             log("calling RANS3PSed2D ctor")
-            self.rans3pf = cRANS3PSed.RANS3PSed2D(
+            self.rans3psed = cRANS3PSed.RANS3PSed2D(
```
[the remark this answers is not in the document]
Comment (Giovanni-Cozzuto-1989) [reply proteus/mprans/RANS3PSed.py]: @adimako 
I'm not sure, only the name was still the fluid one but it was calling in any case the right module and solving the right equations.

File: proteus/mprans/RANS3PSed.py [hunk 2]
```diff
@@ -1701,7 +1702,7 @@ def __init__(self,
                 self.coefficients.angFriction)
         else:
             log("calling  RANS3PSed ctor")
-            self.rans3pf = cRANS3PSed.RANS3PSed(
+            self.rans3psed = cRANS3PSed.RANS3PSed(
```
[the remark this answers is not in the document]
Comment (Giovanni-Cozzuto-1989) [reply proteus/mprans/RANS3PSed.py]: As above

File: proteus/mprans/RANS3PF.py [hunk 3]
```diff
@@ -630,25 +629,29 @@ def initializeMesh(self, mesh):
...
+            #convert from porosity to volume of sediment
+            self.q_vos -= 1.0
+            self.q_vos *= -1.0
         else:
             # TODO make loops faster
             if self.porosityTypes is not None:
-                for eN in range(self.q_porosity.shape[0]):
-                    self.q_porosity[
+                for eN in range(self.q_vos.shape[0]):
+                    self.q_vos[
```
Comment (Giovanni-Cozzuto-1989) [inline proteus/mprans/RANS3PF.py]: I guess this addition is for settting vos from the porosityTypes variable (i.e. from the region basically)

**Pull request merged**